Add extended configuration to rrd4j persistence

J-N-K opened the pull request:
This will add functionality as requested by issue #180 and allow configurable RRD-archives for all datasource types supported by RRD4J.

    # rrd4j:<defname>.def=<dstype>,<heartbeat>,[<min>|U],[<max>|U],<step>
    # rrd4j:<defname>.archives=<consolidationfunction>,<xff>,<steps>,<rows> 
    # rrd4j:<defname>.items=<list of items for this dsname
    rrd4j:ctr5min.def=COUNTER,900,0,U,300
    rrd4j:ctr5min.archives=AVERAGE,0.5,1,365:AVERAGE,0.5,7,300
    rrd4j:ctr5min.items=Item1,Item2

The old defaults will be used if no special configuration can be found. The names for these defintions are default_numeric and default_other and must not be used in own definitions.

buildhive commented:
[openhab » openhab #2872](https://buildhive.cloudbees.com/job/openhab/job/openhab/2872/) FAILURE
Looks like there's a problem with this pull request
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

teichsta commented on bundles/persistence/org.openhab.persistence.rrd4j/OSGI-INF/rrd4j.xml:
```diff
@@ -9,10 +9,12 @@
 
 -->
-<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" name="org.openhab.persistence.rrd4j">
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" activate="activate" deactivate="deactivate" name="org.openhab.persistence.rrd4j">
```
since there is no implementation of active and deactivate adding the methods here can be omitted

teichsta commented on bundles/persistence/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jService.java:
```diff
@@ -65,7 +73,13 @@
+	}
+
+	public void deactivate() {
+	}
```
see above, remove this methods since they seem not to be needed here

teichsta commented:
@J-N-K for this contribution. Please also add the configuration example to openhab_default.cfg and don't forget to squash the commit again finally. Thanks, Thomas E.-E.

buildhive commented:
[openhab » openhab #2873](https://buildhive.cloudbees.com/job/openhab/job/openhab/2873/) FAILURE
Looks like there's a problem with this pull request
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

buildhive commented:
[openhab » openhab #2874](https://buildhive.cloudbees.com/job/openhab/job/openhab/2874/) FAILURE
Looks like there's a problem with this pull request
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

teichsta commented on bundles/persistence/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jService.java:
```diff
@@ -237,56 +249,64 @@ protected synchronized RrdDb getDB(String alias, ConsolFun function) {
+	private ConsolFun getConsolidationFunction(String itemName) {
+		return getRrdDefConfig(itemName).getDefaultConsolFun();
+	}
+*/
```
please remove this commented code

teichsta commented:
sorry, i overlooked the commented code in RRD4jService

buildhive commented:
[openhab » openhab #2875](https://buildhive.cloudbees.com/job/openhab/job/openhab/2875/) FAILURE
Looks like there's a problem with this pull request
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

J-N-K commented:
I followed your suggestions, cleaned up whitespaces a little more and changed the website of rrd4j (they moved from googlecode to github).

Can you explain why buildhive fails?

teichsta commented:
> Can you explain why buildhive fails?

no, it seems to be merge conflict within the Z-Wave code (which should not be related to this change anyhow).

Thanks for incorporating the feedback so quickly.

Best, Thomas E.-E.
